subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -155,6 +167,35 @@ int z_impl_z_zsock_getaddrinfo_internal(const char *host, const char *service,
 		return getaddrinfo_null_host(port, hints, res);
 	}
 
+#define SIN_ADDR(ptr) (((struct sockaddr_in *)(ptr))->sin_addr)
+
+	/* Check for IPv4 numeric address. Start with a quick heuristic check,
+	 * of first char of the address, then do long validating inet_pton()
+	 * call if needed.
+	 */
+	if (RESOLVE_IPV4(family) &&
```

Comment:
Just call the `net_ipaddr_parse()` which will fail if the address is not numeric.

Reply:
> Just call the net_ipaddr_parse() which will fail if the address is not numeric.

Well, it already calls inet_pton() which will fail if address is not numeric. But calling everything in turn is slow, so it applies cheap heuristic checks to speed it up.

Reply:
Ok